monitor/src/ConsoleUI.cpp

```diff
@@ -0,0 +1,96 @@
+// Copyright(c), Philips Medical Systems Nederland B.V., IGT-Devices
+
+#include "ConsoleUI.h"
+#include "Types.h"
+
+#include <format>
+#include <iostream>
+#include <optional>
+#include <sstream>
+#include <string>
+
+namespace {
+
+using Types::IdType;
+using Types::ValueType;
+using Types::Vital;
+using Types::Vitals;
+
+constexpr auto kIdMessage         = "Enter ID (or type 'exit' to quit):\n> ";
+constexpr auto kValueMessage      = "Enter value (or type 'exit' to quit):\n> ";
+constexpr auto kIdErrorMessage    = "Invalid ID\n";
+constexpr auto kValueErrorMessage = "Invalid value\n";
+constexpr auto kRed               = "\033[0;31m";
+constexpr auto kGreen             = "\033[1;32m";
+constexpr auto kReset             = "\033[0m";
+
+template <typename T>
+std::optional<T> tryParse(const std::string& input)
+{
+    std::istringstream iss(input);
+    char               leftover{};
+    if (T value; iss >> value && !(iss >> leftover)) return value;
+    return std::nullopt;
+}
+
+std::string readLine()
+{
+    std::string input;
+    std::getline(std::cin, input);
+    return input;
+}
+
+bool exitRequested(const std::string& input)
+{
+    return input == "exit";
+}
+
+template <typename T>
+std::optional<T> prompt(const std::string& message, const std::string& errorMessage)
+{
+    while (true)
```

Comment:
This function (slightly) exceeds the complexity limit we've set. The number isn't important by itself, but let it trigger the following observations:
1. Semantic distance: The way I would explain this is "prompt till you get a valid input, or user wants to exit". The closest structure may be `do {readline();} while (isInvalid(input))`
2. Return value propagation: To figure out the meaning of `std::nullopt`, I need to look for the caller, and then its caller. What measures could reduce the time to understand this? Explicit value, instead of implicit return? Exception to indicate the intention to abort?